util-hadoop/src/test/java/com/google/cloud/hadoop/util/HadoopCredentialConfigurationTest.java

```diff
@@ -158,15 +155,15 @@ public void metadataServiceIsUsedByDefault() throws Exception {
 
   @Test
   public void applicationDefaultServiceAccountWhenConfigured() throws Exception {
-    environmentVariables.set(CREDENTIAL_ENV_VAR, getStringPath("test-credential.json"));
-
     CredentialFactory credentialFactory = getCredentialFactory();
-
-    GoogleCredentialWithRetry credential =
-        (GoogleCredentialWithRetry) credentialFactory.getCredential(TEST_SCOPES);
-
-    assertThat(credential.getServiceAccountId()).isEqualTo("test-email@gserviceaccount.com");
-    assertThat(credential.getServiceAccountPrivateKeyId()).isEqualTo("test-key-id");
+    withEnvironmentVariable(CREDENTIAL_ENV_VAR, getStringPath("test-credential.json"))
+        .execute(() -> {
```

Comment:
Thank you for your contribution!

One suggestion for future feature - It will be good if `execute` or other method returns result so it can be consumed and/or asserted outside of the lambda - similar pattern to the JUnit `assertThrows` method:
```java
Exception e = assertThrows(Exception.class, () -> ...)
```

Having this functionality will allow to make tests more readable, because it will be obvious what exact commands need env variables to be set.

Reply:
@stefanbirkner FYI

Reply:
Thank you! LGTM